v1/contracts/Store.sol

```diff
@@ -0,0 +1,68 @@
+/*
+  Store implementation.
+ 
+  An implementation of StoreInterface with a fee per second and withdraw functions for the owner.
+*/
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "./StoreInterface.sol";
+import "./Withdrawable.sol";
+import "./FixedPoint.sol";
+
+
+// An implementation of StoreInterface that can accept Oracle fees in ETH or any arbitrary ERC20 token.
+contract Store {
+
+    using SafeMath for uint;
+    using FixedPoint for FixedPoint.Unsigned;
+    using FixedPoint for uint;
+
+    uint private fixedOracleFeePerSecond; // Percentage of 10^18. E.g., 1e18 is 100% Oracle fee.
+    uint private constant FP_SCALING_FACTOR = 10**18;
```

Comment:
Ideally, if we use the `FixedPoint` library for all of our fixed point numbers, we shouldn't need this constant anymore.

Reply:
Hmm I think there is still some scaling factor off somewhere because I ended up with test cases with way too many zeros. I will dig into it